the-collab-lab/tcl-66-smart-shopping-list
# 24. When the user refreshes or navigates around the app, the login page should not flash on the screen. An indictor should show the user that something is happening during wait times. #54

## Description

The` App.jsx` and `useAuth.jsx` are the main files managed here. The `export` method on all components and views needed to be refactored to `default exports`. You will see changes in those files as well. Reacts `lazy` loading will not work on files without default exports.
Lazy loading was added to the Route logic on all views except` Login.jsx`, as well as on the `PrivateRoute.jsx` component. Adding this functionality improves loading time by splitting the code into more manageable bundles, it will only load the component when it is called for by the user.
React's `Suspense` component was added to all views and components in the routing logic, so that a loading indicator is rendered to the user via the `Spinner` component.
The `useAuth` hook that manages the app's authorization state was given an `isAuthorizing `state as well. This state can be used via the hook. In this case in  `App.jsx`. Now, during in app Authorization (specifically when refreshing the page) the same loading indicator will render to the user, letting them know that the app is working. This also eliminates the brief flash we were seeing of the login page on page refresh/ anytime the app authorized the user.

## Related Issue

closes #54 

## Acceptance Criteria

- [x] loading state indicated with spinner during navigation
- [x] authentication state indicated with spinner during page refresh

## Type of Changes

`enhancement`

## Updates

### Before

https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/108297341/e8c4e17e-d770-4baa-a4fc-a8a1ddaf31eb


### After

https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/108297341/c755e394-e81b-4dfa-a54b-5e6149c89850


## Testing Steps / QA Criteria

* git pull from main
* git checkout sc-issue-24
* npm start
* navigate around the app and refresh the page, keep an eye out for flashes of the login page. You should only see the spinner or the page you are navigating to render.
